sdk/storage/azure-storage-blobs/src/blob_responses.cpp

```diff
@@ -58,4 +63,168 @@ namespace Azure { namespace Storage { namespace Blobs {
     };
   }
 
+  void ListBlobContainersPagedResponse::OnNextPage(const Azure::Core::Context& context)
+  {
+    _detail::BlobRestClient::Service::ListBlobContainersSinglePageOptions protocolLayerOptions;
+    protocolLayerOptions.Prefix = m_operationOptions.Prefix;
+    protocolLayerOptions.ContinuationToken = NextPageToken;
+    protocolLayerOptions.MaxResults = m_operationOptions.PageSizeHint;
+    protocolLayerOptions.Include = m_operationOptions.Include;
+    auto response = _detail::BlobRestClient::Service::ListBlobContainersSinglePage(
```

Comment:
You should just be able to call the normal ListBlobContainers here passing in the options after adjusting the continuation token. This means you don't need the friend and no need to reference the protoolLayerOptions or anything in _details.

Also, I don't understand what you're trying to do with the WithReplicaStatus and Context - I do not think you want this. When the page request goes through the pipeline, the policy you already have should just do the right thing.

Reply:
Oh, as I read down, I just discovered what you're doing: You're having ListBlobContainers call OnNextPage internally. Personally, I would not have implemented it this way (although your way works). 

I would have had ListBlobContainers call the protocol layer (just like you did before for XxxSinglePage). Then I'd have OnNextPage call back into ListBlobContainers.  This way, ListBlobContainer is self-contained and does all the work just like all the other XxxClient methods do the work. And then, it just happens to return a PagedResponse instead of a single page result this time. Therefore, your PagedResponse needs the friends and must access _details and other private fields making the code brittle and harder to debug/follow.

Reply:
Yeah, I think it can work. I'll try it tomorrow.